static/app/utils/replays/extractDomNodes.tsx

```diff
@@ -17,187 +13,126 @@ export type Extraction = {
-  if (!frames || !rrwebEvents || rrwebEvents.length < 2 || !finishedAt) {
-    return Promise.reject();
-  }
-
-  return new Promise((resolve, reject) => {
-    const domRoot = document.createElement('div');
-    domRoot.className = 'sentry-block';
-    const {style} = domRoot;
-    style.position = 'fixed';
-    style.inset = '0';
-    style.width = '0';
-    style.height = '0';
-    style.overflow = 'hidden';
-
-    document.body.appendChild(domRoot);
-
-    // Grab the last event, but skip the synthetic `replay-end` event that the
-    // ReplayerReader added. RRWeb will skip that event when it comes time to render
-    const lastEvent = rrwebEvents[rrwebEvents.length - 2];
-
-    const isLastRRWebEvent = (event: RecordingFrame) => lastEvent === event;
-
-    const replayerRef = new Replayer(rrwebEvents, {
-      root: domRoot,
-      loadTimeout: 1,
-      showWarning: false,
-      blockClass: 'sentry-block',
-      speed: 99999,
-      skipInactive: true,
-      triggerFocus: false,
-      plugins: [
-        new BreadcrumbReferencesPlugin({
-          frames,
-          isFinished: isLastRRWebEvent,
-          onFinish: rows => {
-            resolve(rows);
-            setTimeout(() => {
-              if (document.body.contains(domRoot)) {
-                document.body.removeChild(domRoot);
-              }
-            }, 0);
-          },
-        }),
-      ],
-      mouseTail: false,
-    });
-
-    try {
-      // Run the replay to the end, we will capture data as it streams into the plugin
-      replayerRef.pause(finishedAt.getTime());
-    } catch (error) {
-      Sentry.captureException(error);
-      reject(error);
-    }
-  });
-}
+  return new Promise(resolve => {
+    const extractions = new Map<BreadcrumbFrame | SpanFrame, Extraction>();
+    const player = createPlayer(rrwebEvents);
+    let lastEventTimestamp = 0;
+
+    const callback = event => {
+      if (event.type === 2 || event.type === 3) {
+        // Get first frame with a timestamp less than the last seen event timestamp
+        const firstFrameAfterEvent = frames.findIndex(
+          f => f.timestampMs >= lastEventTimestamp
+        );
+
+        lastEventTimestamp = event.timestamp;
+
+        for (let i = firstFrameAfterEvent; i < frames.length; i++) {
+          const frame = frames[i];
+
+          // Sometimes frames have nodeId -1 so we ignore these
+          if (frame.data && 'nodeId' in frame.data && frame.data.nodeId === -1) {
+            continue;
+          }
+
+          // If we found the frame.data.nodeId inside the player at this timestamp, push it to the DOM events list
+          // Otherwise, push with null HTML for now
+          const found = extractNode(frame, player);
+          if (found) {
+            extractions.set(frame, found);
+          } else {
+            extractions.set(frame, {
+              frame,
+              html: null,
+              timestamp: frame.timestampMs,
+            });
+            return;
```

Comment:
I wonder if we should pre-fill `extractions` with this 'empty html' object when it's initialized.

The `return` here worries me a bit today, and is why i'm looking deeper at it. For example: If we have a click in the list, but the click timestamp is after the last rrweb update (maybe the click is a dead click) then this `return` will prevent the `resolve()` call from being made down below.

If we turn the `return` into a continue then the problem should be solved. But then we should still pre-fill all `extractions` so that we handle cases where there are more than one click happening afterwards.